src/vllm_haystack/__init__.py

```diff
@@ -209,7 +214,12 @@ def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union
 
         :param prompt: Prompt text to be sent to the generative model.
         """
-        encoded_prompt = list(self.tokenizer.encode(cast(str, prompt)).ids)
+        encoding = self.tokenizer.encode(cast(str, prompt))
+        if not isinstance(encoding, list):
+            encoded_prompt = list(encoding.ids)
+        else:
+            # the provided tokenizer natively returns the ids from the encode call, such as a Transformers tokenizer
+            encoded_prompt = list(encoding)
```

Comment:
Not pretty, but this should cover 99.99% of cases.

Reply:
Yeah, not thrilled but without writing mapping logic to cover all known possible tokenizers/implementations (such as how `transformers` is using `tokenizers` but returning a different signature on `encode`), wasn't sure of a better way.

Reply:
I changed how this was handled so there's no potential unknown behavior, and an error is raised if there could be.